engines/scc_proxy/lib/scc_proxy/engine.rb

```diff
@@ -263,22 +276,60 @@ def has_no_regcode?(auth_header)
 
         protected
 
+        # rubocop:disable Metrics/PerceivedComplexity
+        # rubocop:disable Metrics/CyclomaticComplexity
+        # rubocop:disable Metrics/AbcSize
         def scc_activate_product
           logger.info "Activating product #{@product.product_string} to SCC"
-          auth = request.headers['HTTP_AUTHORIZATION']
+          auth = nil
+          auth = request.headers['HTTP_AUTHORIZATION'] if request.headers.include?('HTTP_AUTHORIZATION')
+          mode = nil
           if @system.byos?
-            response = SccProxy.scc_activate_product(@product, auth, params[:token], params[:email])
+            mode = 'byos'
+          elsif !@product.free? && @product.extension?
+            regcode_error_message = 'A registration code is required to activate this product'
+            raise ActionController::TranslatedError.new(regcode_error_message) if (@system.payg? || @system.hybrid?) && params[:token].nil?
+
+            # system is not BYOS and
+            # product is a non free extension
+            # and a registration code is provided
+            mode = 'hybrid'
+            # the extensions must be the same version and arch
+            # than base product
+            base_prod = @system.products.find_by(product_type: :base)
+            if base_prod.present? && @product.arch == base_prod.arch && @product.version == base_prod.version
+              request.headers['proxy_byos_mode'] = mode
+              request.headers['scc_login'] = @system.login
+              request.headers['scc_password'] = @system.password
+              request.headers['instance_data'] = params[:instance_data]
+              request.headers['hwinfo'] = params[:hwinfo]
+              response = SccProxy.announce_system_scc(auth, request.headers)
```

Comment:
On further review, I don't think you need `auth` at all...

Consider dropping https://github.com/SUSE/rmt/pull/1198/files#diff-0e5c13122cb7577a624fa4e30eaa3fcfcc0b84a046770be73b96160ba993beb4R284-R285 
... and using here
```
response = SccProxy.announce_system_scc(request.headers['HTTP_AUTHORIZATION'], request.headers)
```

Reply:
... and adding a note to later refactor `SccProxy#announce_system_scc` to not be redundant ;-)